### PR TITLE
Multiple blanx tokens

### PR DESCRIPTION
**Pull request by maciek-pioro**
# Description
Makes it possible for different blanks to be separate tokens and have different embeddings.

# Checklist
   - [x] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [x] Are all variables set with logical defaults that are backward compatible?
   - [x] I attached link to neptune if it was necessary
     - [LINK](https://app.neptune.ai/o/pmtest/org/llm-random/runs/compare?viewId=9aa8f618-f156-40ad-90e8-02d74360ea27&detailsTab=metadata&shortId=LLMRANDOM-3904&dash=charts&type=run&compare=MwNgrMA0CMQ&chartFilter=eval)
   - [x] Have you successfully executed the linter and tests?
   - [x] Are all functions concise and don't require splitting?
   - [x] Is there documentation for arguments and complex logic?
   - [x] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [x] Are all functions placed in appropriate files?
   - [x] Is the PR name meaningful and descriptive?
   - [x] Is PR description provided, or unnecessary?
  
Other questions:
   - [ ] ~If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters~
   - [ ] ~If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)~
   - [ ] ~If there is a need for a second reviewer I requested additional review~

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

File: research/blanks/loss.py [hunk 1]
```diff
@@ -123,9 +126,10 @@ def calculate_llm_loss(
 
     blanks_losses = {}
     if n_blanks > 0:
-        assert blank_id is not None
+        assert len(blanks_ids) > 0
         with torch.no_grad():
-            is_blank = input_tokens.eq(blank_id)
+            # is_blank = input_tokens.isin(blanks)
```
Comment (ciebson123) [inline research/blanks/loss.py:131]: Delete this comment

File: research/blanks/trainer.py [hunk 2]
```diff
@@ -37,6 +37,7 @@ class BlankTrainer:
     max_sequence_length: int
     batch_size: int
     lr_scheduler: AbstractLRScheduler
+    blanks_ids: List[int]
```
Comment (ciebson123) [inline research/blanks/trainer.py:40]: Just a suggestion: Can we make it a tensor? I think then we could make get_is_blank to accept only tensor.
That is just my statically typed brain speaking.

**Review by ciebson123 — COMMENTED** [review]
Great job! I want to see neptune link to some run that proves that it runs.

File: research/blanks/trainer.py (hunk 2 above)
Comment (maciek-pioro) [reply research/blanks/trainer.py:40]: Not a big issue imo

Comment (ciebson123) [reply research/blanks/trainer.py:40]: We decided that it's not important

**Review by ciebson123 — APPROVED** [review]
Looks great!
